text/0000-else.md

```diff
@@ -0,0 +1,219 @@
+
+- [Vue: `v-else` and `v-else-if`](https://vuejs.org/guide/essentials/conditional.html#v-else)
+- [Svelte: `:else` and `:else if`](https://svelte.dev/docs#template-syntax-if)
+- [Angular: `ngIfElse`](https://angular.io/api/common/NgIf)
+- [Solid: `<Show>` and `<Match>`](https://www.solidjs.com/docs/latest/api#%3Cshow%3E)
+- [Aura: `aura:set attribute="else"`](https://developer.salesforce.com/docs/component-library/bundle/aura:if/documentation)
+
+Some other frameworks, such as [React](https://reactjs.org/docs/conditional-rendering.html), [Stencil](https://stenciljs.com/docs/templating-jsx#conditionals), and [Lit](https://lit.dev/docs/templates/conditionals/#conditionals-with-if-statements), have no built-in concept of `else` or `else if`, other than that provided by JavaScript itself.
+
+In the case of LWC, since we already have `if:else` and `if:false`, and since there is no ergonomic way to do conditional rendering in JavaScript (other than `render()`, which requires creating separate template HTML files), it would be most natural to align with the first group of frameworks.
+
+## Detailed design
+
+The basic design is to add three new directives:
+
+- `elseif:true`
+- `elseif:false`
+- `lwc:else`
+
+The reason for using `lwc:else` rather than just `else` is to avoid conflicts with existing attributes that may be named `else`, and to ensure that all directives have a namespace separator (`:`).
+
+### Syntax
+
+All three new directives must be immediately preceded by a sibling `if:true`/`if:false` (or `elseif:true`/`elseif:false`).
+
+```html
+<!-- Valid -->
+<template if:true={foo}></template>
+<template lwc:else></template>
+```
+
+```html
+<!-- Invalid -->
+<template if:true={foo}></template>
+<div>Yolo!</div>
+<template lwc:else></template>
+```
+
+Whitespace is ignored when considering siblinghood.
+
+#### Code comments
+
+In the case of code comments, the behavior depends on whether `lwc:preserve-comments` is enabled or not. When _not_ preserving comments, comments may appear between conditional siblings:
+
+```html
+<!-- Valid -->
+<template>
+    <template if:true={foo}></template>
+    <!-- Comment! -->
+    <template lwc:else></template>
+</template>
+```
+
+Whereas when `lwc:preserve-comments` is enabled, comments become syntactically meaningful and therefore cannot be placed between sibling conditional directives:
+
+```html
+<!-- Invalid -->
+<template lwc:preserve-comments>
+    <template if:true={foo}></template>
+    <!-- Comment! -->
+    <template lwc:else></template>
+</template>
+```
+
+#### Accessors
+
+Similar to `if:true` and `if:false`, the expression passed in to `elseif:true` and `elseif:false` must use [simple dot notation](https://lwc.dev/guide/reference#if%3Atrue%7Cfalse%3D%7Bexpression%7D).
+
+#### Attribute values
+
+To avoid typos (e.g. using `lwc:else` when you mean `lwc:elseif`), `lwc:else` must _not_ have an attribute value. Any attribute value is treated as a compile-time error:

```

Comment:
```suggestion
To avoid typos (e.g. using `lwc:else` when you mean `elseif:true`), `lwc:else` must _not_ have an attribute value. Any attribute value is treated as a compile-time error:

```